src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.DefaultOutputPaths.targets

```diff
@@ -11,29 +11,60 @@ Copyright (c) .NET Foundation. All rights reserved.
     assuming the convention maintained by VS that every Configuration|Platform combination had
     an explicit OutputPath. Since we now want to support leaner project files with less
-    duplication and more automatic defaults, we always set a default OutputPath and can no
-    longer depend on that convention for validation. Getting validation re-enabled with a
-    different mechanism is tracked by https://github.com/dotnet/sdk/issues/350
+    duplication and more automatic defaults, we always set a default OutputPath.
    -->
-  <PropertyGroup>
-    <Configuration Condition="'$(Configuration)'==''">Debug</Configuration>
-    <Platform Condition="'$(Platform)'==''">AnyCPU</Platform>
-    <PlatformName Condition="'$(PlatformName)' == ''">$(Platform)</PlatformName>
 
+  <!-- Projects which don't use Microsoft.NET.Sdk will typically define the OutputPath directly (usually in a
+       Configuration-specific PropertyGroup), so in that case we won't append to it by default. -->
+  <PropertyGroup Condition="'$(UsingNETSdkDefaults)' == 'true'">
+    <AppendTargetFrameworkToOutputPath Condition="'$(AppendTargetFrameworkToOutputPath)' == ''">true</AppendTargetFrameworkToOutputPath>
+  </PropertyGroup>
+
+  <!-- NOTE: If we want to default UseStandardOutputPaths to true when targeting a given version of .NET or higher, this is where we would do it.
+
+    It would look something like this:
+
+  <PropertyGroup Condition="'$(UseStandardOutputPaths)' == '' and
+                            '$(TargetFrameworks)' == '' and
+                            '$(TargetFrameworkIdentifier)' == '.NETCoreApp' and
+                            $([MSBuild]::VersionGreaterThanOrEquals($(TargetFrameworkVersion), 8.0))">
+    <UseStandardOutputPaths>true</UseStandardOutputPaths>
+  </PropertyGroup>
+  -->
+
+  <!-- Handle RootOutputPath and if it wasn't set in Sdk.props -->
+  <PropertyGroup Condition="'$(RootOutputPath)' != ''">
+    <UseStandardOutputPaths Condition="'$(UseStandardOutputPaths)' == ''">true</UseStandardOutputPaths>
+    <BaseStandardOutputPath Condition="'$(BaseStandardOutputPath)' == ''">$(RootOutputPath)</BaseStandardOutputPath>
+  </PropertyGroup>
+
+  <!-- Set BaseOutputPath for standard output format if it wasn't set in Sdk.props -->
+  <PropertyGroup Condition="'$(UseStandardOutputPaths)' == 'true' And '$(BaseOutputPath)' == ''">
+    <BaseStandardOutputPath Condition="'$(BaseStandardOutputPath)' == ''">bin\</BaseStandardOutputPath>
+    <BaseStandardOutputPath>$([MSBuild]::EnsureTrailingSlash($(BaseStandardOutputPath)))</BaseStandardOutputPath>
+
+    <BaseOutputPath>$(BaseStandardOutputPath)build\</BaseOutputPath>
+
+    <!-- If RootOutputPath is set, then projects share the BaseStandardOutputPath, and the project name should be appended to the output path -->
+    <BaseOutputPath Condition="'$(RootOutputPath)' != ''">$(BaseOutputPath)$(MSBuildProjectName)\</BaseOutputPath>
+  </PropertyGroup>
```

Comment:
No, this is for the case where you set properties (which have changed names since this) in your project file instead of in a `Directory.Build.props` file.  So when `Sdk.props` is evaluated, those properties aren't set yet.